### ISSUE TITLE
JWT test WithGoodTokenAsyncFetchFast flaky: Timed out waiting for new connection

### ISSUE THREAD

**Issue by yangminzhu** [envoyproxy/envoy#17303]
*Title*: JWT test WithGoodTokenAsyncFetchFast flaky: Timed out waiting for new connection

*Repro steps*:
With a clean repo at main branch (`56d92a555`), reproduce the flaky with the command:

`
./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.dev //test/extensions/filters/http/jwt_authn:filter_integration_test --runs_per_test=1000  --test_env=ENVOY_IP_TEST_VERSIONS=v4only'
`

Test output:
```
INFO: From Testing //test/extensions/filters/http/jwt_authn:filter_integration_test (run 956 of 1000):
==================== Test output for //test/extensions/filters/http/jwt_authn:filter_integration_test (run 956 of 1000):
[==========] Running 68 tests from 3 test suites.
[----------] Global test environment set-up.
[----------] 28 tests from Protocols/LocalJwksIntegrationTest
[ RUN      ] Protocols/LocalJwksIntegrationTest.WithGoodToken/IPv4_HttpDownstream_HttpUpstream
[       OK ] Protocols/LocalJwksIntegrationTest.WithGoodToken/IPv4_HttpDownstream_HttpUpstream (201 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.WithGoodToken/IPv4_HttpDownstream_Http2Upstream
[       OK ] Protocols/LocalJwksIntegrationTest.WithGoodToken/IPv4_HttpDownstream_Http2Upstream (154 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.WithGoodToken/IPv4_Http2Downstream_HttpUpstream
[       OK ] Protocols/LocalJwksIntegrationTest.WithGoodToken/IPv4_Http2Downstream_HttpUpstream (185 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.WithGoodToken/IPv4_Http2Downstream_Http2Upstream
[       OK ] Protocols/LocalJwksIntegrationTest.WithGoodToken/IPv4_Http2Downstream_Http2Upstream (152 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.ExpiredToken/IPv4_HttpDownstream_HttpUpstream
[       OK ] Protocols/LocalJwksIntegrationTest.ExpiredToken/IPv4_HttpDownstream_HttpUpstream (174 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.ExpiredToken/IPv4_HttpDownstream_Http2Upstream
[       OK ] Protocols/LocalJwksIntegrationTest.ExpiredToken/IPv4_HttpDownstream_Http2Upstream (178 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.ExpiredToken/IPv4_Http2Downstream_HttpUpstream
[       OK ] Protocols/LocalJwksIntegrationTest.ExpiredToken/IPv4_Http2Downstream_HttpUpstream (136 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.ExpiredToken/IPv4_Http2Downstream_Http2Upstream
[       OK ] Protocols/LocalJwksIntegrationTest.ExpiredToken/IPv4_Http2Downstream_Http2Upstream (171 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.MissingToken/IPv4_HttpDownstream_HttpUpstream
[       OK ] Protocols/LocalJwksIntegrationTest.MissingToken/IPv4_HttpDownstream_HttpUpstream (161 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.MissingToken/IPv4_HttpDownstream_Http2Upstream
[       OK ] Protocols/LocalJwksIntegrationTest.MissingToken/IPv4_HttpDownstream_Http2Upstream (151 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.MissingToken/IPv4_Http2Downstream_HttpUpstream
[       OK ] Protocols/LocalJwksIntegrationTest.MissingToken/IPv4_Http2Downstream_HttpUpstream (157 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.MissingToken/IPv4_Http2Downstream_Http2Upstream
[       OK ] Protocols/LocalJwksIntegrationTest.MissingToken/IPv4_Http2Downstream_Http2Upstream (152 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.ExpiredTokenHeadReply/IPv4_HttpDownstream_HttpUpstream
[       OK ] Protocols/LocalJwksIntegrationTest.ExpiredTokenHeadReply/IPv4_HttpDownstream_HttpUpstream (174 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.ExpiredTokenHeadReply/IPv4_HttpDownstream_Http2Upstream
[       OK ] Protocols/LocalJwksIntegrationTest.ExpiredTokenHeadReply/IPv4_HttpDownstream_Http2Upstream (152 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.ExpiredTokenHeadReply/IPv4_Http2Downstream_HttpUpstream
[       OK ] Protocols/LocalJwksIntegrationTest.ExpiredTokenHeadReply/IPv4_Http2Downstream_HttpUpstream (144 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.ExpiredTokenHeadReply/IPv4_Http2Downstream_Http2Upstream
[       OK ] Protocols/LocalJwksIntegrationTest.ExpiredTokenHeadReply/IPv4_Http2Downstream_Http2Upstream (143 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.NoRequiresPath/IPv4_HttpDownstream_HttpUpstream
[       OK ] Protocols/LocalJwksIntegrationTest.NoRequiresPath/IPv4_HttpDownstream_HttpUpstream (145 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.NoRequiresPath/IPv4_HttpDownstream_Http2Upstream
[       OK ] Protocols/LocalJwksIntegrationTest.NoRequiresPath/IPv4_HttpDownstream_Http2Upstream (173 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.NoRequiresPath/IPv4_Http2Downstream_HttpUpstream
[       OK ] Protocols/LocalJwksIntegrationTest.NoRequiresPath/IPv4_Http2Downstream_HttpUpstream (146 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.NoRequiresPath/IPv4_Http2Downstream_Http2Upstream
[       OK ] Protocols/LocalJwksIntegrationTest.NoRequiresPath/IPv4_Http2Downstream_Http2Upstream (159 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.CorsPreflight/IPv4_HttpDownstream_HttpUpstream
[       OK ] Protocols/LocalJwksIntegrationTest.CorsPreflight/IPv4_HttpDownstream_HttpUpstream (179 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.CorsPreflight/IPv4_HttpDownstream_Http2Upstream
[       OK ] Protocols/LocalJwksIntegrationTest.CorsPreflight/IPv4_HttpDownstream_Http2Upstream (150 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.CorsPreflight/IPv4_Http2Downstream_HttpUpstream
[       OK ] Protocols/LocalJwksIntegrationTest.CorsPreflight/IPv4_Http2Downstream_HttpUpstream (158 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.CorsPreflight/IPv4_Http2Downstream_Http2Upstream
[       OK ] Protocols/LocalJwksIntegrationTest.CorsPreflight/IPv4_Http2Downstream_Http2Upstream (140 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.FilterStateRequirement/IPv4_HttpDownstream_HttpUpstream
[       OK ] Protocols/LocalJwksIntegrationTest.FilterStateRequirement/IPv4_HttpDownstream_HttpUpstream (156 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.FilterStateRequirement/IPv4_HttpDownstream_Http2Upstream
[       OK ] Protocols/LocalJwksIntegrationTest.FilterStateRequirement/IPv4_HttpDownstream_Http2Upstream (151 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.FilterStateRequirement/IPv4_Http2Downstream_HttpUpstream
[       OK ] Protocols/LocalJwksIntegrationTest.FilterStateRequirement/IPv4_Http2Downstream_HttpUpstream (172 ms)
[ RUN      ] Protocols/LocalJwksIntegrationTest.FilterStateRequirement/IPv4_Http2Downstream_Http2Upstream
[       OK ] Protocols/LocalJwksIntegrationTest.FilterStateRequirement/IPv4_Http2Downstream_Http2Upstream (165 ms)
[----------] 28 tests from Protocols/LocalJwksIntegrationTest (4481 ms total)

[----------] 28 tests from Protocols/RemoteJwksIntegrationTest
[ RUN      ] Protocols/RemoteJwksIntegrationTest.WithGoodToken/IPv4_HttpDownstream_HttpUpstream
[       OK ] Protocols/RemoteJwksIntegrationTest.WithGoodToken/IPv4_HttpDownstream_HttpUpstream (193 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.WithGoodToken/IPv4_HttpDownstream_Http2Upstream
[       OK ] Protocols/RemoteJwksIntegrationTest.WithGoodToken/IPv4_HttpDownstream_Http2Upstream (175 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.WithGoodToken/IPv4_Http2Downstream_HttpUpstream
[       OK ] Protocols/RemoteJwksIntegrationTest.WithGoodToken/IPv4_Http2Downstream_HttpUpstream (176 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.WithGoodToken/IPv4_Http2Downstream_Http2Upstream
[       OK ] Protocols/RemoteJwksIntegrationTest.WithGoodToken/IPv4_Http2Downstream_Http2Upstream (181 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.FetchFailedJwks/IPv4_HttpDownstream_HttpUpstream
[       OK ] Protocols/RemoteJwksIntegrationTest.FetchFailedJwks/IPv4_HttpDownstream_HttpUpstream (211 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.FetchFailedJwks/IPv4_HttpDownstream_Http2Upstream
[       OK ] Protocols/RemoteJwksIntegrationTest.FetchFailedJwks/IPv4_HttpDownstream_Http2Upstream (197 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.FetchFailedJwks/IPv4_Http2Downstream_HttpUpstream
[       OK ] Protocols/RemoteJwksIntegrationTest.FetchFailedJwks/IPv4_Http2Downstream_HttpUpstream (207 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.FetchFailedJwks/IPv4_Http2Downstream_Http2Upstream
[       OK ] Protocols/RemoteJwksIntegrationTest.FetchFailedJwks/IPv4_Http2Downstream_Http2Upstream (191 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.FetchFailedMissingCluster/IPv4_HttpDownstream_HttpUpstream
[2021-07-12 23:40:33.757][59157][error][filter] [source/extensions/filters/http/common/jwks_fetcher.cc:46] fetch: fetch pubkey [uri = https://pubkey_server/pubkey_path] failed: [cluster = pubkey_cluster] is not configured
[       OK ] Protocols/RemoteJwksIntegrationTest.FetchFailedMissingCluster/IPv4_HttpDownstream_HttpUpstream (156 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.FetchFailedMissingCluster/IPv4_HttpDownstream_Http2Upstream
[2021-07-12 23:40:33.910][59233][error][filter] [source/extensions/filters/http/common/jwks_fetcher.cc:46] fetch: fetch pubkey [uri = https://pubkey_server/pubkey_path] failed: [cluster = pubkey_cluster] is not configured
[       OK ] Protocols/RemoteJwksIntegrationTest.FetchFailedMissingCluster/IPv4_HttpDownstream_Http2Upstream (153 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.FetchFailedMissingCluster/IPv4_Http2Downstream_HttpUpstream
[2021-07-12 23:40:34.073][59299][error][filter] [source/extensions/filters/http/common/jwks_fetcher.cc:46] fetch: fetch pubkey [uri = https://pubkey_server/pubkey_path] failed: [cluster = pubkey_cluster] is not configured
[       OK ] Protocols/RemoteJwksIntegrationTest.FetchFailedMissingCluster/IPv4_Http2Downstream_HttpUpstream (164 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.FetchFailedMissingCluster/IPv4_Http2Downstream_Http2Upstream
[2021-07-12 23:40:34.219][59370][error][filter] [source/extensions/filters/http/common/jwks_fetcher.cc:46] fetch: fetch pubkey [uri = https://pubkey_server/pubkey_path] failed: [cluster = pubkey_cluster] is not configured
[       OK ] Protocols/RemoteJwksIntegrationTest.FetchFailedMissingCluster/IPv4_Http2Downstream_Http2Upstream (156 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetch/IPv4_HttpDownstream_HttpUpstream
[       OK ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetch/IPv4_HttpDownstream_HttpUpstream (192 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetch/IPv4_HttpDownstream_Http2Upstream
[       OK ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetch/IPv4_HttpDownstream_Http2Upstream (178 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetch/IPv4_Http2Downstream_HttpUpstream
[       OK ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetch/IPv4_Http2Downstream_HttpUpstream (210 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetch/IPv4_Http2Downstream_Http2Upstream
[       OK ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetch/IPv4_Http2Downstream_Http2Upstream (193 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetchFast/IPv4_HttpDownstream_HttpUpstream
[       OK ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetchFast/IPv4_HttpDownstream_HttpUpstream (174 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetchFast/IPv4_HttpDownstream_Http2Upstream
[       OK ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetchFast/IPv4_HttpDownstream_Http2Upstream (185 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetchFast/IPv4_Http2Downstream_HttpUpstream
[       OK ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetchFast/IPv4_Http2Downstream_HttpUpstream (198 ms)
[ RUN      ] Protocols/RemoteJwksIntegrationTest.WithGoodTokenAsyncFetchFast/IPv4_Http2Downstream_Http2Upstream
[2021-07-12 23:40:45.722][56656][critical][assert] [test/integration/http_integration.cc:470] assert failure: 0. Details: Timed out waiting for new connection.
[2021-07-12 23:40:45.722][56656][critical][backtrace] [./source/server/backtrace.h:104] Caught Aborted, suspect faulting address 0x582370000dd50
[2021-07-12 23:40:45.722][56656][critical][backtrace] [./source/server/backtrace.h:91] Backtrace (use tools/stack_decode.py to get line numbers):
[2021-07-12 23:40:45.722][56656][critical][backtrace] [./source/server/backtrace.h:92] Envoy version: 0/1.19.0-dev/test/DEBUG/BoringSSL
[2021-07-12 23:40:45.785][56656][critical][backtrace] [./source/server/backtrace.h:96] #0: Envoy::SignalAction::sigHandler() [0x530ec9c]
[2021-07-12 23:40:45.785][56656][critical][backtrace] [./source/server/backtrace.h:96] #1: __restore_rt [0x7f6213505980]
[2021-07-12 23:40:45.836][56656][critical][backtrace] [./source/server/backtrace.h:96] #2: Envoy::HttpIntegrationTest::waitForNextUpstreamRequest() [0x2caeb54]
[2021-07-12 23:40:45.886][56656][critical][backtrace] [./source/server/backtrace.h:96] #3: Envoy::HttpIntegrationTest::waitForNextUpstreamRequest() [0x2cad558]
[2021-07-12 23:40:45.936][56656][critical][backtrace] [./source/server/backtrace.h:96] #4: Envoy::Extensions::HttpFilters::JwtAuthn::(anonymous namespace)::RemoteJwksIntegrationTest_WithGoodTokenAsyncFetchFast_Test::TestBody() [0x2b38844]
[2021-07-12 23:40:45.986][56656][critical][backtrace] [./source/server/backtrace.h:96] #5: testing::internal::HandleSehExceptionsInMethodIfSupported<>() [0x67b31c4]
[2021-07-12 23:40:46.037][56656][critical][backtrace] [./source/server/backtrace.h:96] #6: testing::internal::HandleExceptionsInMethodIfSupported<>() [0x67a23ab]
[2021-07-12 23:40:46.085][56656][critical][backtrace] [./source/server/backtrace.h:96] #7: testing::Test::Run() [0x678e193]
[2021-07-12 23:40:46.133][56656][critical][backtrace] [./source/server/backtrace.h:96] #8: testing::TestInfo::Run() [0x678eb67]
[2021-07-12 23:40:46.180][56656][critical][backtrace] [./source/server/backtrace.h:96] #9: testing::TestSuite::Run() [0x678f31c]
[2021-07-12 23:40:46.228][56656][critical][backtrace] [./source/server/backtrace.h:96] #10: testing::internal::UnitTestImpl::RunAllTests() [0x679b2bf]
[2021-07-12 23:40:46.275][56656][critical][backtrace] [./source/server/backtrace.h:96] #11: testing::internal::HandleSehExceptionsInMethodIfSupported<>() [0x67b6f94]
[2021-07-12 23:40:46.323][56656][critical][backtrace] [./source/server/backtrace.h:96] #12: testing::internal::HandleExceptionsInMethodIfSupported<>() [0x67a455b]
[2021-07-12 23:40:46.370][56656][critical][backtrace] [./source/server/backtrace.h:96] #13: testing::UnitTest::Run() [0x679ae63]
[2021-07-12 23:40:46.417][56656][critical][backtrace] [./source/server/backtrace.h:96] #14: RUN_ALL_TESTS() [0x4fbd301]
[2021-07-12 23:40:46.464][56656][critical][backtrace] [./source/server/backtrace.h:96] #15: Envoy::TestRunner::RunTests() [0x4fbcbb0]
[2021-07-12 23:40:46.512][56656][critical][backtrace] [./source/server/backtrace.h:96] #16: main [0x4fbadde]
[2021-07-12 23:40:46.512][56656][critical][backtrace] [./source/server/backtrace.h:96] #17: __libc_start_main [0x7f6213123bf7]
```

cc @qiwzhang

**Comment by qiwzhang** [comment]
OK,  please assign it to me.

**Comment by snowp** [comment]
Fixed by #17306

**Comment by qiwzhang** [comment]
Please close this issue.  It should have been fixed.
